File: model/src/main/java/com/blackopalsolutions/goneviral/model/response/GetUserResponse.java
package com.blackopalsolutions.goneviral.model.response;

import com.blackopalsolutions.goneviral.model.domain.User;

public class GetUserResponse extends Response {
    private final User user;

    public GetUserResponse(User user) {
        super(true);
        this.user = user;
    }

    public GetUserResponse(String message) {
        super(message);
        this.user = null;
    }

    public User getUser() {
        return user;
    }
}

File: shared/src/commonMain/kotlin/com/blackopalsolutions/goneviral/model/response/GetUserResponse.kt
package com.blackopalsolutions.goneviral.model.response

import com.blackopalsolutions.goneviral.model.domain.User

class GetUserResponse : Response {
    val user: User?

    constructor(user: User) : super(true) {
        this.user = user
    }

    constructor(message: String) : super(message) {
        user = null
    }
}